File: Practice Questions/Building Logic/CPP/Fibonacci.cpp
//Fibonacci sequence is a mathematical series in which each number is the sum of the two preceding ones, usually starting with 0 and 1.
// The sequence goes: 0, 1, 1, 2, 3, 5, 8, 13, 21, and so on.
// F(n) = F(n-1) + F(n-2), where F(0) = 0 and F(1) = 1

#include <iostream> 
using namespace std; //fibonacci sequence 
int Fib(int n){ 
    if(n==0){ 
        return 0; 
    }
    if(n==1){ 
        return 1; 
    }
    return Fib(n-1)+Fib(n-2); 
} 
int main(){ 
    int n; 
    cout<<"Enter your Number:\n"; 
    cin>>n; 
    cout<<Fib(n); 
    return 0; 
}

//code for fibonacci sequence using recursion.

File: Practice Questions/Building Logic/Python/Fibonacci.py
n=int(input("Enter the number:"))
def fib(n):
    if(n==0):
        return 0
    elif(n==1):
        return 1
    return fib(n-1)+fib(n-2)

seq=[]
for i in range(0,n+1):
    seq.append(fib(i))
print(f"Fibonacci number of {n} is: {fib(n)}")
print(f"The Fibonacci sequence till n numbers is:")
for i in seq:
    print(i)